ql/cashflows/inflationcoupon.cpp

```diff
@@ -62,11 +62,16 @@ namespace QuantLib {
 
 
     Rate InflationCoupon::rate() const {
+        calculate();
+        return rate_;
+    }
+
+    void InflationCoupon::performCalculations() const {
         QL_REQUIRE(pricer_, "pricer not set");
         // we know it is the correct type because checkPricerImpl checks on setting
         // in general pricer_ will be a derived class, as will *this on calling
         pricer_->initialize(*this);
```

Comment:
Can this be moved into the constructor when `pricer_ != nullptr`?

Reply:
I think we can't because a pricer can be shared between coupons

Reply:
OK, but just to check, is it ok for different coupons to initialize the same pricer potentially multiple times?

Reply:
Yes that is what happens today too, usually

Reply:
Yes that is what happens today too, usually